src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -40,25 +40,33 @@ namespace ripple {
 NotTEC
 preflight0(PreflightContext const& ctx)
 {
-    uint32_t const nodeNID = ctx.app.config().NETWORK_ID;
-    std::optional<uint32_t> const txNID = ctx.tx[~sfNetworkID];
-
-    if (nodeNID <= 1024)
+    if (isPseudoTx(ctx.tx))
     {
-        // legacy networks have IDs 1024 and below. These networks cannot
-        // specify NetworkID in txn
-        if (txNID)
-            return telNETWORK_ID_MAKES_TX_NON_CANONICAL;
+        // all emitted and pseudo transactions are free to pass, do not need
+        // network id
```

Comment:
What should we do if a pseudo transaction has a network id? Right now it's ignored, which is probably fine. But some other options would be:
1) If present, it needs to follow the same rules as other transactions (i.e. match the configured network id), fail if the network is less than 1024.
2) Fail if present in any form.
3) Assert if present in any form, but allow to continue processing in release mode.

I'd probably prefer (3). If someone does add a network id to a pseudo transaction, and they did it intentionally, the assert will let them know this code needs to change. But you're call if you want to add it or not. I'm fine either way.

As a side note, I also prefer not to have the empty `if` block, but that's just a preference. I'm also fine if you want to keep it as-is.

Reply:
I'd vote for option 1. It has the advantage of being simple to add that check. Simply change the check to
```
if (isPseudoTx(ctx.tx) && !ctx.tx.isFieldPresent(sfNetworkID))
```

Reply:
I like option 1 because its simple.

Reply:
@dangell7 I don't think that patch is going to work If the network ID is required, but isn't present the new patch let's the transaction succeed.

Reply:
fixed in [d8b2ba4](https://github.com/XRPLF/rippled/pull/4737/commits/d8b2ba42a6b2ba1a327ecb7f8021bdff662f3e75)